### ISSUE TITLE
[bug/question?] Consul Connect Vault Root CA rotation

### ISSUE THREAD

**Issue by sgtang** [hashicorp/consul#10643]
#### Overview of the Issue

We've been using Consul Connect with Vault as an external CA, with Consul given a token with permissions to completely manage the Vault PKI endpoints (create, update, delete). A couple days ago, the Root CA created by Consul expired, and was not rotated or renewed. We do not know if this is intended behavior or a bug. Is it expected for the user to manually rotate to new PKI endpoints before the root CA expires, even if Consul is given permissions to manage these endpoints? Thanks.

#### Reproduction Steps

Steps to reproduce this issue, eg:

1. Create a Consul cluster with any number of servers with Connect enabled
2. Configure an external Vault CA
3. Wait for the root CA to expire

### Operating system and Environment details

Ubuntu 18.04, Consul 1.9.6, Envoy 1.16.4

### Log Fragments

No applicable logs, besides seeing expected 'certificate validation' errors on proxy debug logs, and Consul server logs showing the successful rotation of the Vault periodic token.

**Comment by dnephin** [comment]
Thank you for reporting this issue!

I had a look over our docs (https://www.consul.io/docs/connect/ca/vault#configuration, and https://www.consul.io/docs/agent/options#vault-ca-provider-ca_provider-vault) to see what they had to say, and it looks like we don't really provide enough information on those pages.

Consul will never rotate the root CA, and while looking into this, I discovered it does indeed mount the PKI path using a `max_lease_ttl` of 365 days: https://github.com/hashicorp/consul/blob/v1.10.1/agent/connect/ca/provider_vault.go#L177-L179

We don't have any way to rotate this certificate, but we should at least document this `max_lease_ttl`. Ideally we would also allow it to be configured, so that it could be set to something greater than 1 year. We use intermediate certs for signing (and those intermediate certs can be rotated), so a longer TTL on a root cert should be ok. 

I created #10762 to track the work to make those improvements.

I should also mention #9924 added a new metric for monitoring this expiration. That will be available in Consul 1.11, and should make it possible to alert when the CA is close to expiry.
